src/Documentation/Action/DocumentationAction.php

```diff
@@ -56,25 +65,31 @@ public function __construct(ResourceNameCollectionFactoryInterface $resourceName
 
             return;
         }
+
         $this->formatsProvider = $formatsProvider;
     }
 
-    public function __invoke(Request $request = null): Documentation
+    public function __invoke(Request $request = null): DocumentationInterface
     {
         if (null !== $request) {
-            $context = ['base_url' => $request->getBaseUrl(), 'spec_version' => $request->query->getInt('spec_version', $this->swaggerVersions[0] ?? 2)];
+            $context = ['base_url' => $request->getBaseUrl(), 'spec_version' => $request->query->getInt('spec_version', $this->swaggerVersions[0] ?? 3)];
             if ($request->query->getBoolean('api_gateway')) {
                 $context['api_gateway'] = true;
             }
             $request->attributes->set('_api_normalization_context', $request->attributes->get('_api_normalization_context', []) + $context);
 
             $attributes = RequestAttributesExtractor::extractAttributes($request);
         }
+
         // BC check to be removed in 3.0
         if (null !== $this->formatsProvider) {
             $this->formats = $this->formatsProvider->getFormatsFromAttributes($attributes ?? []);
         }
 
+        if (null !== $this->openApiFactory && isset($context) && 3 === $context['spec_version']) {
```

Comment:
Initialize the context to an empty array and use the null coalescing?

Reply:
Context is declared when request isn't null above, I don't want to change this existing behavior.

Reply:
Why not?